### ISSUE TITLE
Feedback: upload progress

### ISSUE THREAD

**Issue by adojaan** [gphotosuploader/gphotos-uploader-cli#185]
Thank you very much for this piece of software. After moving away from Amazon photos (downloading, renaming), it helps to create Google albums. For me the installation had some minor problems but genrally it worked fine. 
I have couple of questions/suggestions regarding the workflow.

- can I configure on-screen logging somehow? I tested with ~1Gb of files (1241 files) from one folder, specifying it in config.hjson. Then launched the script with push argument. It found all the images as expected and gave [info] about it:
```
...
[info]   Found item to upload: /home/kristjan/Windir/Pictures/Amazon/Amazon Photos Downloads/2004/villemi sp 04/2004 villemi sp 04 IMG_5601.JPG
[info]   1241 files pending to be uploaded in folder '/home/kristjan/Windir/Pictures/Amazon/Amazon Photos Downloads/2004'.
```
...but then - silence. No indication of progress. I would suggest to give log entry for every file uploaded and every album created, to understand the status. It has run several hours now and the only way to see the progress, is to check Google Photos website (where I see that photos are added). But as the files are not uploaded in particular order it is hard to understand, if all are loaded or some are still waiting. The only indication of activity of the script so far is some errors about album creation
```
[error]  Album creation failed: name=jogevaaed2004mai, error=googleapi: Error 500: Internal error encountered., backendError
[error]  Album creation failed: name=juuni2004 QM suvepäevad, error=googleapi: Error 500: Internal error encountered., backendError
[error]  Album creation failed: name=juuni2004 QM suvepäevad, error=googleapi: Error 500: Internal error encountered., backendError
```
Actually, the albums were created and filled with photos, so the errors are probably solved.

- Should the job finally end and provide some final notice of uploaded files, or I just have to cancel it?

- How to parse (and grep, etc) logs in uploads.db or it is meant for script's own internal use only?

- and general question - is it supposed to run as a service so that I can configure multiple jobs (according to my photolocations) in config.hjson and it will upload photos whenever they are added to the folders or I should modfy configuration file each time new folder is added and launch it then?

**Comment by adojaan** [comment]
Now next error was displayed, unfortunately google quota had exceeded... 
```
[error]  error while trying to create this media item, err=too many retries: file=/home/kristjan/Windir/Pictures/Amazon/Amazon Photos Downloads/2004/vaibla_aug04/2004 vaibla_aug04 MVI_6709.AVI, err=googleapi: Error 429: Quota exceeded for quota metric 'All requests' and limit 'All requests per day' of service 'photoslibrary.googleapis.com' for consumer 'project_number:579313930041'., rateLimitExceeded
[fail] X Error processing /home/kristjan/Windir/Pictures/Amazon/Amazon Photos Downloads/2004/vaibla_aug04/2004 vaibla_aug04 MVI_6709.AVI
[error]  Album creation failed: name=vaibla_aug04, error=googleapi: Error 429: Quota exceeded for quota metric 'All requests' and limit 'All requests per day' of service 'photoslibrary.googleapis.com' for consumer 'project_number:579313930041'., rateLimitExceeded
```
From Quotas page https://console.cloud.google.com/iam-admin/quota I see that the quota is 10000 and it is fully used. Seems that every upload calls api several times (10?) - othewise uploading of 1241 files shouldn't consume so many requests. I understand that media upload and album creating and adding to album are probably separate requests. Would it be possible to optimize it somehow? 

It could be useful to mention the quota problem on your project page along with estimate of how many photos per day could be possibly uploaded.

Now I cancelled the script and try again tomorrow. I hope it is able to resume automatically :)

**Comment by pacoorozco** [comment]
Hi @adojaan 

First of all, thanks a lot for all your feedback, It really helps a lot. I've seen several improvements points that I'll try to tackle soon. There are some questions that are already fixed, so let me answer it:
> can I configure on-screen logging somehow?
Yes, you can use `--debug` and you are going to find a lot of information. Thanks to your comment I'll convert some `debug` messages into `info` ones.

> Should the job finally end and provide some final notice of uploaded files, or I just have to cancel it?
The CLI should end and show you a summary of the number of uploaded files. You should wait until the end (hopefully).

> How to parse (and grep, etc) logs in uploads.db or it is meant for script's own internal use only?
It's intended for internal use only. **What would you like to do with its content?**

> s it supposed to run as a service so that I can configure multiple jobs (according to my photolocations) in config.hjson and it will upload photos whenever they are added to the folders or I should modfy configuration file each time new folder is added and launch it then?
No, it's not a service. It's a CLI to run it time to time. Every time you run it, the CLI will lookr for the files to be uploaded and afterwards it will upload it. 

That means, for example, that if you add some files while the CLI is running, the `gphotos-uploader-cli` will not be aware until you run it again.

You can configure several folders (with the same target account or a different one) and `gphotos-uploader-cli` will review it sequentially.

I hope these answers can help you to understand better the CLI, do not hesitate to ask for anything that is not clear enough. Thanks again for the complete feedback :tada:

**Comment by adojaan** [comment]
Dear Paco,

thank you for the response, I will try to use --debug flag for more output. And yes, I can confirm, that in the end of upload summary was shown. 
My idea of parsing the log was just to understand the progress of file uploading. Something like cat log|grep uploaded|wc, or tail -f log.

I understand that in some cases configuration of the path in hson file would be useful, i.e. if I have a huge photos folder and add subfolders of photos from particular events, then launch cli. In my use case I have archive of photos, organized by years and events. I modify them (check and change exif dates, batch rename files to match folder names as well, batch rename folders to include years, if they dont have yet, or convert years from 05 to 2005, etc), then upload whole year's folder or some of its subfolders (if they are big). I do it year by year. So first I understood that each time I have to change the path in configuration file.  My idea was to propose uploading from current dir, but only now I realised that it already seems to work. Although the documentation says it must be full path. If I leave the source folder empty or use relative path "./" in configuration file and then launch ... push from desired folder, it reports that I have 0 files pending in that directory (that is true).  I don't now if it is on purpose or random or if it messes something up in logging for resuming. I would have proposed exactly the same: if folder is left empty in config.hjson, it should use current directory.  Or take it from cli flag like --in ./myfolder. Whenever the --in flag is used, its argument overrides config.hjson setting and if flag is used without an argument, ./ is used for it.

One thing I did not understand (and did not find from your github documentation) is about passphrase. Each time I run push it asks twice "Enter passphrase to unlock /home/kristjan/.gphotos-uploader-cli:" , I just press Enter twice to get past of it. What's the meaning of it or how to avoid it?

Emerging from my use case, I have some ideas (that may help to "make gphotos-uploader-cli the best tool" :) ) that are relevant in context of archiving old photos to Google. These ideas are mainly related to analysing photo collection before upload (finding missing or suspicious exif dates, finding and converting incopatible file formats, renaming files, etc). But it is probably not relevant to most of your users who upload just new present-day photos that have all the data already correct. Therefore they might be out of your scope.  Anyway, let me know if you are interested, then I can give a better structured description of what I mean.

Best regards, Kristjan
